contracts/exchangeIssuance/ExchangeIssuanceLeveraged.sol

```diff
@@ -0,0 +1,926 @@
+     * @param _exchange              Enum indicating which exchange to use
+     * @param _paymentParams         Encoded payment data used to get the minimum output amount
+     * @param _originalSender        Address of the original sender to return the eth to
+     */
+    function _liquidateLongTokensForETH(address _longTokenUnderlying, uint256 _amountToReturn, Exchange _exchange, address _originalSender, bytes memory _paymentParams) internal returns(uint256) {
+            uint256 minAmountOutputToken = _decodePaymentParams(_paymentParams);
+            uint256 ethAmount = _swapLongForOutputToken(_longTokenUnderlying, _amountToReturn, WETH, _exchange);
+            require(ethAmount >= minAmountOutputToken, "ExchangeIssuance: INSUFFICIENT OUTPUT AMOUNT");
+            if (ethAmount > 0) {
+                IWETH(WETH).withdraw(ethAmount);
+                (payable(_originalSender)).sendValue(ethAmount);
+            }
+            return ethAmount;
+    }
+
+
+    /**
+     * Obtains the tokens necessary to return the flashloan by swapping the short tokens obtained
+     * from issuance and making up the shortfall using the users funds.
+     *
+     * @param _longTokenUnderlying   Underlying token of the collateral AToken, which is the token to be returned
+     * @param _amountRequired        Amount of longTokenUnderlying required to repay the flashloan
+     * @param _setToken          Address of the SetToken to be issued
+     * @param _setAmount         Amount of SetTokens to issue
+     * @param _originalSender    Address of the user who initiated the redemption
+     * @param _exchange          Exchange to use for swap
+     * @param _paymentToken      Enum controlling what token the user will receive
+     * @param _paymentParams     Encoded parameters for given payment method
+     */
+    function _obtainLongTokens(
+        address _longTokenUnderlying,
+        uint256 _amountRequired,
+        ISetToken _setToken,
+        uint256 _setAmount,
+        address _originalSender,
+        Exchange _exchange,
+        PaymentToken _paymentToken,
+        bytes memory _paymentParams
+    ) internal {
+        uint longTokenObtained = _swapShortForLongTokenUnderlying(_setToken, _setAmount, _longTokenUnderlying, _exchange);
+        uint longTokenShortfall = _amountRequired.sub(longTokenObtained);
+        uint amountInputToken;
+        address inputToken;
+        if(_paymentToken == PaymentToken.ERC20){
+            (inputToken, amountInputToken) = _makeUpShortfallWithERC20(_longTokenUnderlying, longTokenShortfall, _exchange, _originalSender, _paymentParams);
+        }
+        else {
+            amountInputToken = _makeUpShortfallWithETH(_longTokenUnderlying, longTokenShortfall, _exchange, _originalSender, _paymentParams);
+            inputToken = WETH;
+        }
+        emit ExchangeIssue(_originalSender, _setToken, IERC20(inputToken), amountInputToken, _setAmount);
```

Comment:
Actually it would be "WETH" which is even worse since you couldn't differenciate an ERC20 issuance for WETH from issuing for ETH.

Changed it to set inputToken to `ETH_ADDRESS` in the else clause